src/commands/install-modules.yml

```diff
@@ -0,0 +1,70 @@
+description: >
+  Install your node modules with automated caching and best practices applied.
+
+parameters:
+  pkg-manager:
+    type: enum
+    enum: ["npm", "yarn"]
+    default: "npm"
+    description: Select the default node package manager to use.
+  app-dir:
+    type: string
+    default: "~/project"
+    description: Path to the directory containing your package.json file. Not needed if package.json lives in the root.
+  override-ci:
+    type: boolean
+    default: false
+    description: |
+      By default, modules will be installed with "npm ci" or "yarn install --frozen-lockfile", this override will use the standard "install" command.
+  with-cache:
+    type: boolean
+    default: true
+    description: Enable automatic caching of your node dependancies for increased speed.
+  cache-version:
+    type: string
+    default: v1
+    description: Change the default cache version if you need to clear the cache for any reason.
+  cache-key:
+    type: string
+    default: package.json
```

Comment:
This should be the lock file "package-lock.json". There's cases where `package.json` can change but the lock file won't. The lock file itself is what determines what specific versions of all packages will get installed.

Reply:
I know this is the standard theory but I have run into issues in the past doing this. I am not sure why but the lockfile appears to change checksum any time you run npm install. I am not sure why but, the package.json file should be as stateless. While the lockfile specifies what version to install, the package.json file is only acted upon manually and thus can be modified unless you intentionally change it.

Reply:
This is what the lock file is for. If it changes, a dependency **has** changed. It works the same in Go for `go.mod` vs `go.sum` and in PHP for `composer.json` and `composer.lock`.

`package.json` isn't suitable for a cache checksum because it can be non-deterministic. You can list an npm package in there such as `react` which today (03/27/2020) would install React v16.13.1. Then, the cache would store that version of React. In two months, React could have a new minor release, v16.14.0. The project on GitHub would still load `v16.13.1` from the cache (because package.json hasn't changed) but when you run `npm install` React v16.14.0 would install. Now the cache doesn't match reality and this isn't as effective.

The whole point of the lock file is that it's deterministic. A lock file will cause `npm install` to always install the same version of every single package unless a flag is passed to update the lockfile and/or `npm update` is run.